src/overlays/actors/ovl_En_Gb2/z_en_gb2.c

```diff
@@ -46,98 +50,983 @@ const ActorInit En_Gb2_InitVars = {
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Gb2/func_80B0F660.s")
+void func_80B0FE7C(GlobalContext* globalCtx) {
+    globalCtx->nextEntranceIndex = 0x9C20;
+    globalCtx->unk_1887F = 0x40;
+    gSaveContext.nextTransition = 0x40;
+    globalCtx->sceneLoadFlag = 0x14;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Gb2/func_80B0F6DC.s")
+void func_80B0FEBC(EnGb2* this, GlobalContext* globalCtx) {
+    if ((globalCtx->msgCtx.unk1202A == 3) && (globalCtx->msgCtx.unk1202E == 7)) {
+        globalCtx->msgCtx.unk1202A = 4;
+        gSaveContext.eventInf[4] |= 0x80;
+        this->unk_26E = 0x14D1;
+        this->unk_288 = 10;
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Gb2/func_80B0F728.s")
+    if (func_800B84D0(&this->actor, globalCtx)) {
+        func_801518B0(globalCtx, this->unk_26E, &this->actor);
+        this->actionFunc = func_80B0FFA8;
+    } else if ((this->actor.xzDistToPlayer < 300.0f) || this->actor.isTargeted) {
+        func_800B863C(&this->actor, globalCtx);
+    }
+}
+
+void func_80B0FFA8(EnGb2* this, GlobalContext* globalCtx) {
+    u8 temp_v0 = func_80152498(&globalCtx->msgCtx);
+
+    if (temp_v0 == 5) {
+        if (func_80147624(globalCtx)) {
+            if (this->unk_26C & 2) {
+                globalCtx->msgCtx.unk11F22 = 0x43;
+                globalCtx->msgCtx.unk12023 = 4;
+                this->unk_26E = 0x14D1;
+                this->unk_288 = 30;
+                gSaveContext.eventInf[4] &= (u8)~0x80;
+                this->unk_26C &= ~2;
+                this->actionFunc = func_80B0FEBC;
+            } else {
+                this->unk_26E = func_80B0F7FC(this);
+                func_801518B0(globalCtx, this->unk_26E, &this->actor);
+            }
+        }
+    } else if ((temp_v0 == 4) && func_80147624(globalCtx)) {
+        if (this->unk_26E == 0x14D5) {
+            switch (globalCtx->msgCtx.choiceIndex) {
+                case 0:
+                    if (gSaveContext.rupees < this->unk_288) {
+                        play_sound(NA_SE_SY_ERROR);
+                        this->unk_26E = 0x14D7;
+                        this->unk_26C |= 2;
+                        func_801518B0(globalCtx, this->unk_26E, &this->actor);
+                    } else {
+                        func_8019F208();
+                        this->unk_26E = 0x14D8;
+                        func_801518B0(globalCtx, this->unk_26E, &this->actor);
+                    }
+                    break;
+
+                case 1:
+                    func_8019F230();
+                    this->unk_26E = 0x14D6;
+                    this->unk_26C |= 2;
+                    func_801518B0(globalCtx, this->unk_26E, &this->actor);
+                    break;
+            }
+        } else if (this->unk_26E == 0x14DA) {
+            switch (globalCtx->msgCtx.choiceIndex) {
+                case 0:
+                    func_8019F208();
+                    func_801159EC(this->unk_288 * -1);
```

Comment:
Out of curiosity, would this match?
```suggestion
                    func_801159EC(-this->unk_288);
```

Reply:
I thought this would be change codegen so I didn't actually try, but yeah it works no problem.